source/fundamentals/promises.txt

```diff
@@ -0,0 +1,147 @@
+======================
+Promises and Callbacks
+======================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecols
+
+Overview
+--------
+
+NodeJS driver uses the asynchronous Javascript API to communicate with
+your MongoDB cluster.
+
+Asynchronous Javascript allows you to execute operations without waiting for 
+the processing thread to become free. This helps prevent your application
+from becoming unresponsive to web UI rendering or network requests when
+executing long-running operations. For more information about asynchronous 
+Javascript, see the MDN web documentation on 
+`Asynchronous Javascript <https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous>`_.
+
+This section describes two features of asynchronous Javascript -- 
+`Promises`_ and `Callbacks`_ -- that you can use with the NodeJS driver to
+access the results of your method calls to your MongoDB cluster.
+
+Promises
+--------
+
+A Promise is an object returned by the asynchronous method call that allows
+you to access information on the eventual success or failure of the operation.
+For more information on Promises and related terminology, see the MDN
```

Comment:
Same as the above comment on "for more information"

Feel free to ignore this comment if you disagree with it. I'm partial to included the links within the sentence because it cuts down on the bulk of the doc, and achieves the same purpose.